src/Processors/Executors/StreamingFormatExecutor.cpp

```diff
@@ -47,18 +51,43 @@ void StreamingFormatExecutor::setQueryParameters(const NameToNameMap & parameter
         values_format->setQueryParameters(parameters);
 }
 
-size_t StreamingFormatExecutor::execute(ReadBuffer & buffer)
+void StreamingFormatExecutor::preallocateResultColumns(size_t num_bytes, const Chunk & chunk)
+{
+    if (!try_preallocate)
+        return;
+
+    try_preallocate = false; /// do it once
+
+    if (total_bytes && num_bytes && total_chunks > 2)
```

Comment:
You're only using the amount of chunks here, right?
Why `total_chunks > 2` and not `total_chunks > 1`?

Reply:
'++factor' is gone in [79dc763](https://github.com/ClickHouse/ClickHouse/pull/74945/commits/79dc7637ac64c1d3b5f1bae9b82332574ef0d6b9)  until I am missing something.

Reply:
> Why total_chunks > 2 and not total_chunks > 1

Ok , saying 'always' lets mean this. I'll fix it.

Reply:
You're not missing anything. It's just that I thought I had pushed my review comments when I wrote the comment https://github.com/ClickHouse/ClickHouse/pull/74945#issuecomment-2775031278, but apparently they did not get published. So now https://github.com/ClickHouse/ClickHouse/pull/74945#discussion_r2027171632 got mixed with obsolete comments

Reply:
>You suggest making decision if we plan to preallocate in constructror and don't introduce this attribute?

Sorry for the confusion. Maybe I didn't explain myself clearly. What I meant with my

>why did you use the amount of bytes instead of just the total number of chunks? I'd expect factor to be the total chunks if the first chunk is representative.

was to use `total_chunks` as the factor because I think it's simpler to understand (and doesn't require to pass num_bytes in a few methods), not to use it to distinguish whether or not we need to preallocate. However, I agree that  depending on the nature of the data using bytes might be a better approach.

Anyway, I think you could use your code as it was before (without the `total_chunks`) and preallocate always.